go/oasis-node/cmd/debug/fixgenesis/fixgenesis.go

```diff
@@ -0,0 +1,219 @@
+	signature.Signed
+}
+
+func doFixGenesis(cmd *cobra.Command, args []string) {
+	if err := cmdCommon.Init(); err != nil {
+		cmdCommon.EarlyLogAndExit(err)
+	}
+
+	// Load the old genesis document.
+	f := flags.GenesisFile()
+	raw, err := ioutil.ReadFile(f)
+	if err != nil {
+		logger.Error("failed to open genesis file",
+			"err", err,
+		)
+		os.Exit(1)
+	}
+
+	// Parse as the old format.  At some point all the important things
+	// will be versioned, but this is not that time.
+	var oldDoc oldDocument
+	if err = json.Unmarshal(raw, &oldDoc); err != nil {
+		logger.Error("failed to parse old genesis file",
+			"err", err,
+		)
+		os.Exit(1)
+	}
+
+	// Actually fix the genesis document.
+	newDoc, err := updateGenesisDoc(&oldDoc)
+	if err != nil {
+		logger.Error("failed to fix genesis document",
+			"err", err,
+		)
+		os.Exit(1)
+	}
+
+	// Validate the new genesis document.
+	if err = newDoc.SanityCheck(); err != nil {
+		logger.Error("new genesis document sanity check failed",
+			"err", err,
+		)
+		os.Exit(1)
+	}
+
+	// Write out the new genesis document.
+	w, shouldClose, err := cmdCommon.GetOutputWriter(cmd, cfgNewGenesis)
+	if err != nil {
+		logger.Error("failed to get writer for fixed genesis file",
+			"err", err,
+		)
+		os.Exit(1)
+	}
+	if shouldClose {
+		defer w.Close()
+	}
+	if raw, err = json.Marshal(newDoc); err != nil {
+		logger.Error("failed to marshal fixed genesis document into JSON",
+			"err", err,
+		)
+		os.Exit(1)
+	}
+	if _, err = w.Write(raw); err != nil {
+		logger.Error("failed to write new genesis file",
+			"err", err,
+		)
+		os.Exit(1)
+	}
+}
+
+func updateGenesisDoc(oldDoc *oldDocument) (*genesis.Document, error) {
+	// Create the new genesis document template.
+	newDoc := &genesis.Document{
+		Height:     oldDoc.Height,
+		Time:       oldDoc.Time,
+		ChainID:    oldDoc.ChainID,
+		EpochTime:  oldDoc.EpochTime,
+		RootHash:   oldDoc.RootHash,
+		Staking:    oldDoc.Staking,
+		KeyManager: oldDoc.KeyManager,
+		Scheduler:  oldDoc.Scheduler,
+		Beacon:     oldDoc.Beacon,
+		Consensus:  oldDoc.Consensus,
+		HaltEpoch:  oldDoc.HaltEpoch,
+		ExtraData:  oldDoc.ExtraData,
+	}
+
+	// This currently is entirely registry genesis state changes.
+	oldReg, newReg := oldDoc.Registry, newDoc.Registry
+
+	// First copy the registry things that have not changed.
+	newReg.Parameters = oldReg.Parameters
+	newReg.Entities = oldReg.Entities
+	newReg.Runtimes = oldReg.Runtimes
+	newReg.SuspendedRuntimes = oldReg.SuspendedRuntimes
+	newReg.NodeStatuses = oldReg.NodeStatuses
+
+	// The node descriptor signature envelope format in the registry has
+	// changed.  Convert to the new envelope.
+	//
+	// Note: Actually using the genesis document requires that some
+	// signature checks be disabled.
+	for _, osn := range oldReg.Nodes {
+		var nsn node.MultiSignedNode
+		nsn.MultiSigned.Signatures = []signature.Signature{osn.Signed.Signature}
+
+		// TODO: Someone that understands the issue can also fix the node
```

Comment:
We just need to clear all flags and set the (new) validator flag (other nodes are not persisted anyway AFAIK).